app-backend/tool/src/main/scala/ast/package.scala

```diff
@@ -87,4 +87,16 @@ package object ast extends MapAlgebraCodec {
     def min(other: MapAlgebraAST): MapAlgebraAST.Operation =
       MapAlgebraAST.Min(List(self, other), UUID.randomUUID(), generateMetadata)
   }
+
+  implicit class FocalBuffer(val self: MapAlgebraAST) {
```

Comment:
I could say that quote argues in favour of putting that logic back into `MapAlgebraAST` ;)